### PR TITLE
DOC improvements - SelectField

### LINKED ISSUE #123: [SelectField] Add async options and menu item icon examples
See discord [discussion](https://discord.com/channels/920755200552226868/1168230830271893626/1172179237667614820)

### PR DESCRIPTION
**Pull request by jycouet**
Closes #123

Adding 
- [x] with icons
- [x] async

### CODE REVIEW

File: packages/svelte-ux/src/routes/docs/components/SelectField/+page.svelte [hunk 1]
```diff
@@ -136,6 +160,33 @@
...
+          option === selected && 'font-semibold',
+          option.group ? 'px-4' : 'px-2'
+        )}
+        scrollIntoView={index === highlightIndex}
+      >
+        <div class="flex gap-2">
+          <Icon path={option.icon} />
+          <div>{option.name}</div>
+        </div>
+      </MenuItem>
```
[the remark this answers is not in the document]
Comment (techniq) [reply packages/svelte-ux/src/routes/docs/components/SelectField/+page.svelte:185]: FWIW, `shadcn/ui` calls our `SelectField` [Combobox](https://ui.shadcn.com/docs/components/combobox), and has the search as a field within the menu, and not as an input.  They also have [Dropdown Menu](https://ui.shadcn.com/docs/components/dropdown-menu) which is more akin to `MenuField`.

Supporting dividers and submenus has been on my list to add at some point.

There is also [Menubar](https://ui.shadcn.com/docs/components/menubar) and [Navigation menu](https://ui.shadcn.com/docs/components/navigation-menu) and maybe a few others if I keep looking :)

File: packages/svelte-ux/src/routes/docs/components/SelectField/+page.svelte [hunk 2]
```diff
@@ -102,6 +107,25 @@
...
+<Preview>
+  <SelectField options={optionsAsync} {loading} />
+  <Button
+    on:click={() => {
+      loading = true;
+      // simulate async loading with a loading of 2 seconds!
+      setTimeout(() => {
+        optionsAsync = optionsAsync.length === 0 ? options : [];
+        loading = false;
+      }, 2000);
```
Comment (techniq) [inline packages/svelte-ux/src/routes/docs/components/SelectField/+page.svelte:121]: Should unloading happen immediately?

File: packages/svelte-ux/src/routes/docs/components/SelectField/+page.svelte [hunk 3]
```diff
@@ -102,6 +107,25 @@
...
+    }}
+    {loading}
+  >
+    {optionsAsync.length === 0 ? 'Load options' : 'unload options'}
```
Comment (techniq) [inline packages/svelte-ux/src/routes/docs/components/SelectField/+page.svelte:125]: nit: "Unload options"

File: packages/svelte-ux/src/routes/docs/components/SelectField/+page.svelte (hunk 2 above)
Comment (jycouet) [reply packages/svelte-ux/src/routes/docs/components/SelectField/+page.svelte:121]: It's an example... So it doesn't really matter. Butt ok 😜

File: packages/svelte-ux/src/routes/docs/components/SelectField/+page.svelte (hunk 1 above)
[the remark this answers is not in the document]
Comment (techniq) [reply packages/svelte-ux/src/routes/docs/components/SelectField/+page.svelte]: Sorry, I have a tendency to add too much context that can confuse the primary task :).  Looks good to me, I'll mention the above to @willnationsdev in #127 as it's more topical there.

**Pull request merged**